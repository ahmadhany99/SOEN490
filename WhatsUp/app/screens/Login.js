/* eslint-disable react/no-unescaped-entities */
import React from "react";
import { useState } from "react";
import { StyleSheet, Text, View, Image, Alert } from "react-native";
import AppButton from "../components/AppButton";
import AppTextInput from "../components/AppTextInput";
import Links from "../components/Links";
import Screen from "../components/Screen";
import TitleHeaders from "../components/TitleHeaders";
import logo from "../Images/w3.png";
import { auth, db } from "../firebase";
import { collection, getDocs, query, where } from "firebase/firestore";
import { signInWithEmailAndPassword } from "firebase/auth";
import { useNavigation } from "@react-navigation/native";
import colors from "../config/colors";
import { SimpleLineIcons } from "@expo/vector-icons";

export default function Login() {
  const navigation = useNavigation();
  const [email, setEmail] = useState("jasmine@organizer.com");
  const [password, setPassword] = useState("capstone123");
  const [isOrganizer] = useState(false);
  const handleLogin = async () => {
    await signInWithEmailAndPassword(auth, email, password)
      .then(async () => {
        const q = query(collection(db, "users"), where("email", "==", email));
        const querySnapshot = await getDocs(q);
        if (querySnapshot != null) {
          querySnapshot.forEach((doc) => {
            if (isOrganizer == doc.data().isOrganizer) {
              navigation.navigate("Attendee");
              setPassword("");
            } else {
              navigation.navigate("Organizer");
              setPassword("");
            }
          });
        }
      })
      .catch(() => {
        Alert.alert("Try again", "Invalid email or password.");
      });
  };
  const SignupPressed = async () => {
    navigation.navigate("SignUp");
  };

  const ResetPassword = () => {
    navigation.navigate("ResetPassword");
  };
  return (
    <Screen
      resizeMode='cover'
      backgroundImage='https://cdn.dribbble.com/users/479289/screenshots/4521207/media/c3e3bd246bbff91101a54e69daa8b1f0.gif'
      style={{ paddingHorizontal: "10%" }}
    >
      <Image
        source={logo}
        style={{ marginTop: 18, width: 166, height: 212, alignSelf: "center" }}
      />
      <TitleHeaders
        style={{ fontSize: 38, marginTop: 20, marginBottom: 5 }}
        title='Login'
        isTitle={true}
      />
      <AppTextInput
        placeholder='Email'
        keyboardType='email-address'
        onChangeText={(text) => setEmail(text.toLowerCase())}
        value={email}
        style={{
          shadowColor: "black", // IOS
          shadowOffset: { height: 1, width: 1 }, // IOS
          shadowOpacity: 0.2, // IOS
          shadowRadius: 2, //IOS
          elevation: 4, // Android
        }}
      />
      <AppTextInput
        placeholder='Password'
        secureTextEntry
        onChangeText={(text) => setPassword(text)}
        value={password}
        style={{
          shadowColor: "black", // IOS
          shadowOffset: { height: 1, width: 1 }, // IOS
          shadowOpacity: 0.2, // IOS
          shadowRadius: 2, //IOS
          elevation: 4, // Android
        }}
      />
      <View style={styles.organizertwo}>
        <Links
          style={styles.link}
          link='Forgot Password?'
          onPress={ResetPassword}
        />
      </View>
      <AppButton title='Login' onPress={handleLogin} />
      <View style={{ flexDirection: "row", justifyContent: "center" }}>
        <Text style={styles.text}>Don't have an account? </Text>
        <Links style={styles.link} link='Sign up' onPress={SignupPressed} />
      </View>
      <View
        style={{
          flexDirection: "row",
          alignItems: "center",
          paddingVertical: 15,
        }}
      >
        <View style={{ flex: 1, height: 1, backgroundColor: "lightgrey" }} />
        <View>
          <Text
            style={{
              width: 105,
              textAlign: "center",
              color: "darkgray",
              fontSize: 14,
            }}
          >
            Or login with
          </Text>
        </View>
        <View style={{ flex: 1, height: 1, backgroundColor: "lightgrey" }} />
      </View>
      <View style={styles.rowContainer}>
        <SimpleLineIcons
          name='social-facebook'
          size={30}
          style={{
            color: colors.secondary,
            fontSize: 30,
            alignSelf: "center",
            paddingVertical: 10,
          }}
          onPress={() => console.log("Facebook")}
          testID='facebook'
        />
        <SimpleLineIcons
          name='social-twitter'
          size={30}
          style={{
            color: colors.secondary,
            fontSize: 30,
            paddingHorizontal: 40,
            alignSelf: "center",
          }}
          onPress={() => console.log("Facebook")}
          testID='twitter'
        />
        <SimpleLineIcons
          name='social-google'
          size={30}
          style={{
            color: colors.secondary,
            fontSize: 30,
            paddingVertical: 10,
          }}
          onPress={() => console.log("Facebook")}
          testID='google'
        />
      </View>
    </Screen>
  );
}

const styles = StyleSheet.create({
  organizer: {
    alignItems: "flex-start",
    width: "50%",
  },
  organizertwo: {
    alignItems: "flex-end",
    width: "98%",
  },
  rowContainer: {
    flexDirection: "row",
    alignSelf: "center",
  },
  roundshape: {
    backgroundColor: "grey",
    height: 65,
    width: 65,
    justifyContent: "center",
    alignSelf: "center",
    alignContent: "center",
    borderRadius: 45,
  },
  text: {
    color: colors.darkerGrey,
    fontSize: 16,
  },
  text2: {
    fontSize: 16,
    color: colors.darkGrey,
  },
});
