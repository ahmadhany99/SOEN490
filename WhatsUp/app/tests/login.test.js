import "react-native";
import React from "react";
import renderer from "react-test-renderer";
import Login from "../screens/Login";
import { NavigationContainer } from "@react-navigation/native";
import {
  render,
  waitFor,
  fireEvent,
  screen,
} from "@testing-library/react-native";
jest.useFakeTimers();
it("renders correctly", () => {
  const tree = render(
    <NavigationContainer>
      <Login />
    </NavigationContainer>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test("Login sucessfully", async () => {
  render(
    <NavigationContainer>
      <Login />
    </NavigationContainer>
  );
  await waitFor(() =>
    fireEvent.changeText(
      screen.getByPlaceholderText("Email"),
      "hanyelbob99@gmail.com"
    )
  );
  await waitFor(() =>
    fireEvent.changeText(screen.getByPlaceholderText("Password"), "capstone123")
  );
  await waitFor(() => fireEvent.press(screen.getAllByText("Login")[1]));
});
