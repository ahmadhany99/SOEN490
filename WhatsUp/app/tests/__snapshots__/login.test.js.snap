// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "paddingTop": 20,
      },
      Object {
        "marginTop": 5,
        "padding": 10,
      },
    ]
  }
>
  <Image
    source={1}
    style={
      Object {
        "alignSelf": "center",
        "height": 212,
        "width": 166,
      }
    }
  />
  <Text
    style={
      Object {
        "fontSize": 28,
        "fontWeight": "bold",
        "marginLeft": "1%",
      }
    }
  >
    Login
  </Text>
  <Text
    style={
      Object {
        "color": "#747474",
        "marginBottom": 6,
        "marginLeft": "1%",
        "paddingVertical": 6,
      }
    }
  >
    Please enter your details
  </Text>
  <View
    style={
      Array [
        Object {
          "borderColor": "#989797",
          "borderRadius": 7,
          "borderStyle": "solid",
          "borderWidth": 1,
          "marginLeft": "1%",
          "marginVertical": 10,
          "paddingHorizontal": 12,
          "paddingVertical": 7,
          "width": "98%",
        },
        undefined,
      ]
    }
  >
    <TextInput
      keyboardType="email-address"
      onChangeText={[Function]}
      placeholder="Email"
      style={
        Object {
          "color": "black",
          "fontFamily": "Helvetica Neue",
          "fontSize": 18,
        }
      }
      value=""
    />
  </View>
  <View
    style={
      Array [
        Object {
          "borderColor": "#989797",
          "borderRadius": 7,
          "borderStyle": "solid",
          "borderWidth": 1,
          "marginLeft": "1%",
          "marginVertical": 10,
          "paddingHorizontal": 12,
          "paddingVertical": 7,
          "width": "98%",
        },
        undefined,
      ]
    }
  >
    <TextInput
      onChangeText={[Function]}
      placeholder="Password"
      secureTextEntry={true}
      style={
        Object {
          "color": "black",
          "fontFamily": "Helvetica Neue",
          "fontSize": 18,
        }
      }
      value=""
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "width": "98%",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <Text
        style={
          Object {
            "color": "#00c0a4",
            "fontSize": 16,
            "fontWeight": "400",
          }
        }
      >
        Forgot Password?
      </Text>
    </View>
  </View>
  <View
    accessibilityState={
      Object {
        "disabled": false,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#00c0a4",
        "borderRadius": 7,
        "elevation": 4,
        "justifyContent": "center",
        "marginLeft": "1%",
        "marginVertical": 40,
        "opacity": 1,
        "padding": 12,
        "shadowColor": "black",
        "shadowOffset": Object {
          "height": 1,
          "width": 1,
        },
        "shadowOpacity": 0.2,
        "shadowRadius": 3,
        "width": "98%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FFF",
          "fontFamily": "Helvetica Neue",
          "fontSize": 18,
          "fontWeight": "bold",
        }
      }
    >
      Login
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#484545",
          "fontSize": 16,
        }
      }
    >
      Don't have an account? 
    </Text>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <Text
        style={
          Object {
            "color": "#00c0a4",
            "fontSize": 16,
            "fontWeight": "400",
          }
        }
      >
        Sign up
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "paddingVertical": 15,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "lightgrey",
          "flex": 1,
          "height": 1,
        }
      }
    />
    <View>
      <Text
        style={
          Object {
            "color": "darkgray",
            "fontSize": 14,
            "textAlign": "center",
            "width": 105,
          }
        }
      >
        Or login with
      </Text>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "lightgrey",
          "flex": 1,
          "height": 1,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignSelf": "center",
        "flexDirection": "row",
      }
    }
  >
    <Text />
    <Text />
    <Text />
  </View>
</View>
`;
